internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml

```diff
@@ -137,67 +130,50 @@ spec:
         set -x
         if [ "$(params.hotfix)" == "true" ]; then
             echo "Hotfix build"
-            fbcOptIn="false"
+            mustOverwriteFromIndexImage="false"
             mustSignIndexImage="true"
             mustPublishIndexImage="true"
         elif [ "$(params.stagedIndex)" == "true" ]; then
             echo "Staged Index build"
-            fbcOptIn="false"
+            mustOverwriteFromIndexImage="false"
             mustSignIndexImage="false"
             mustPublishIndexImage="false"
         else
             # check if this fbc fragment is opt-in
             echo "Fetching the image bundle from $(params.fbcFragment)..."
             PULL_SPEC_LIST=$(opm render "$(params.fbcFragment)" | jq -r \
             'select(.schema == "olm.bundle") | "\(.image)" | split("@")[0]' |uniq)
+
+            fbcOptIn="true"
             for PULL_SPEC in ${PULL_SPEC_LIST}; do
                 # make sure they query is done using the internal name instead of the public
                 PULL_SPEC="${PULL_SPEC//registry.redhat.io/registry.access.redhat.com}"
-
                 echo "Attempting to fetch from ${FETCH_URL} to check if fragment is \`fbc_opt_in==true\`..."
-                # shellcheck disable=SC2207
-                #   (warning): Prefer mapfile or read -a to split command output (or quote to avoid splitting).
-                # shellcheck disable=SC2006 # (style): Use \$(...) notation instead of legacy backticks `...`.
-                # shellcheck disable=SC2086 # (info): Double quote to prevent globbing and word splitting.
-                fbcOptIn+=(`isfbcOptIn ${PULL_SPEC}`)
+                if [ "$(isFBCOptIn "${PULL_SPEC}")" = "false" ]; then
+                  fbcOptIn="false"
+                  break
+                fi
             done
-
-            # shellcheck disable=SC2207
-            #   (warning): Prefer mapfile or read -a to split command output (or quote to avoid splitting).
-            # shellcheck disable=SC2068 # (error): Double quote array expansions to avoid re-splitting elements.
-            fbcOptIn=($(printf "%s\n" ${fbcOptIn[@]} |sort |uniq))
-
-            # shellcheck disable=SC2128 # (warning): Expanding an array without an index only gives the first element.
+            mustOverwriteFromIndexImage="${fbcOptIn}"
```

Comment:
Ok, so you basically renamed `fbcOptIn` to `mustOverwriteFromIndexImage`. Wouldn't it make sense to use it even here from the start? 

```
...
else
  mustOverwriteFromIndexImage=true
  for ...
    if [ "$(isFBCOptIn "${PULL_SPEC}")" = "false" ]; then
      mustOverwriteFromIndexImage=false
```

Reply:
I think this will actually make it clearer - there are potentially multiple images checked, so with the different name, we are saying: if isFBCOptIn is false for an image, we need to override the index image.

Reply:
I changed this again so we can keep the fbc_opt_in as it was found in the image as  per our conversation  [here](https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1733483922183879?thread_ts=1733474423.638719&cid=C04PZ7H0VA8). 
 
- The IsFBCOptIn check is back top, so hotfix and stage would also check the flag.
- the fbcOptIn value is not replaced anymore.
 
 Can you PTAL?

Reply:
Ah, I see, I didn't realize there were two different things - one is the result and one is overwrite_from_index parameter for the iib call. Yep, it makes sense now.